.github/workflows/build.yml

```diff
@@ -0,0 +1,45 @@
+name: "Build"
+on:
+    push:
+      branches:
+        - main
+    pull_request:
+
+jobs:
+  build:
+    name: Build Job ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        java: ["11"]
+    
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
```

Comment:
Check if there's newer versions of those actions, I know there's `actions/setup-java@v3` already at least.